### PR TITLE
Alerting: Use VirtualizedSelect for ContactPointSelector

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR improves the performance when having a long list of contact points in the alert rule form ( using simplified routing).
The PR conditionally uses virtualized window when we have more than 500 contact points. Virtualized select does not support descriptions for options, so we prefer keeping this descriptions for users that use few number of contact points.

For testing it, I provisioned 5k contact points. (see attached video)

**Why do we need this feature?**

We need a UI that behaves smoothly when using a long list of contact points.

**Who is this feature for?**

All users.

**Special notes for your reviewer:**

https://github.com/grafana/grafana/assets/33540275/56a3b19c-938a-4067-89f3-9270a80071b9

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx [hunk 1]
```diff
@@ -80,8 +81,7 @@ export function ContactPointSelector({
             render={({ field: { onChange, ref, ...field }, fieldState: { error } }) => (
               <>
                 <div className={styles.contactPointsSelector}>
-                  <Select
-                    {...field}
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx:84]: there is no need for passing these props  as the `onchange` is the one handling the changes in a custom way, and typescript was complaining about it.

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx [hunk 2]
```diff
@@ -42,9 +42,10 @@ export function ContactPointSelector({
   const contactPointInForm = watch(`contactPoints.${alertManager}.selectedContactPoint`);
 
   const selectedContactPointWithMetadata = options.find((option) => option.value.name === contactPointInForm)?.value;
-  const selectedContactPointSelectableValue = selectedContactPointWithMetadata
-    ? { value: selectedContactPointWithMetadata, label: selectedContactPointWithMetadata.name }
-    : undefined;
+  const selectedContactPointSelectableValue: SelectableValue<ContactPointWithMetadata> =
+    selectedContactPointWithMetadata
+      ? { value: selectedContactPointWithMetadata, label: selectedContactPointWithMetadata.name }
+      : { value: undefined, label: '' };
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx:48]: needed for using the generic type in `VirtualizedSelect`

**Comment by gillesdemey** [comment]
I don't think the virtualized select component supports a `description` so in terms of UX this feels like a step back sadly :(

image

However I'm not opposed to this change if we have a lot of users with such a high amount of contact points.

**Review by gillesdemey — COMMENTED** [review]
We should be able to pass `virtualized={shouldVirtualize}` to the regular `Select` component so we don't need to dynamically switch to rendering another component :)

**Review by gillesdemey — APPROVED** [review]
🚀

**Pull request merged**